src/GNDStk/gnds-tree-visit.hpp

```diff
@@ -0,0 +1,126 @@
+
+// -----------------------------------------------------------------------------
+// Definitions:
+//    Tree::meta()
+//    Tree::child()
+//    Tree::operator()()
+// -----------------------------------------------------------------------------
+
+// ------------------------
+// Tree::meta()
+// ------------------------
+
+// string
+template<
+   template<class...> class MCON,
+   template<class...> class CCON
+>
+inline decltype(auto) Tree<MCON,CCON>::meta(const std::string &str) const
```

Comment:
I'm really struggling with so many methods being defined outside of their classes. It's difficult for me to know the context of the code. Besides, I also think there is a lot of duplicated code because you have to write the template parameters over and over. Am I missing something?

Reply:
I think things follow the style guide pretty closely now.